File: frontend/src/types/packing.ts
export type CautionType = "decomposable" | "fragile" | "waste" | "edible" | "none";

export interface BoxInput {
  id: string;
  length: number;
  width: number;
  height: number;
  caution: CautionType;
}

export interface ContainerInput {
  length: number;
  width: number;
  height: number;
}

export interface PackingRequest {
  container: ContainerInput;
  boxes: BoxInput[];
}

export interface BoxPlacement extends BoxInput {
  x: number;
  y: number;
  z: number;
}

export interface PackingResponse {
  placements: BoxPlacement[];
  utilization: number;
  result_id: number;
  report_url: string;
}

File: backend/app/schemas/packing.py
from pydantic import BaseModel, Field
from typing import List, Literal

CautionType = Literal["decomposable", "fragile", "waste", "edible", "none"]

class BoxInput(BaseModel):
    id: str
    length: float
    width: float
    height: float
    caution: CautionType

class ContainerInput(BaseModel):
    length: float
    width: float
    height: float

class PackingRequest(BaseModel):
    container: ContainerInput
    boxes: List[BoxInput]

class BoxPlacement(BaseModel):
    id: str
    x: float
    y: float
    z: float
    length: float
    width: float
    height: float
    caution: CautionType

class PackingResponse(BaseModel):
    placements: List[BoxPlacement]
    utilization: float
    result_id: int
    report_url: str
